pkg/terraform/diagnose.go

```diff
@@ -78,4 +78,9 @@ var conditions = []condition{{
 
 	reason:  "GCPComputeBackendTimeout",
 	message: `GCP is experiencing backend service interuptions, the compute instance failed to create in reasonable time.`,
+}, {
+	match: regexp.MustCompile(`Error: could not contact API: .*`),
+
+	reason:  "BaremetalIronicAPITimeout",
+	message: `Timed out connecting to Ironic API. Please make sure that Ironic API service is up and running.`,
```

Comment:
I think we should not assume a user knows what Ironic is (we try to hide it as much as possible), and give them some more details:

*Timed out waiting for provisioning service. This failure can be caused by misconfiguration or inability to download the machine operating system images. Please check the bootstrap host for failing services.*

Coupled with https://github.com/openshift/installer/pull/3156, users will get a list of services to check when connecting to the bootstrap.

Reply:
Awesome. This makes a lot of sense. Will update.